src/main/java/org/jabref/gui/maintable/MainTable.java

```diff
@@ -217,17 +216,12 @@ public void paste() {
         List<BibEntry> entriesToAdd = Globals.clipboardManager.extractData();
 
         if (!entriesToAdd.isEmpty()) {
-            // Add new entries
-            NamedCompound ce = new NamedCompound((entriesToAdd.size() > 1 ? Localization.lang("paste entries") : Localization.lang("paste entry")));
+            database.getDatabase().insertEntries(entriesToAdd);
+            // TODO because of this line, when implementing batch undo insert entries, model constructor after batch remove entries with a boolean for paste
+            undoManager.addEdit(new UndoableInsertEntries(database.getDatabase(), entriesToAdd, true));
```

Comment:
We should avoid `Boolean` parameters. See https://java.by-comparison.com/.

Please introduce an enum `AdditionMode` with values `PASTE`, `INSERT`. With that way, it is crystal clear what `paste = false` means. In other context `paste = false` could mean `cut` (or `copy`), so with an enum is really clear what semantics is intended.

Reply:
What is the `TODO` here? 😇